tests/baselines/reference/thisInArrowFunctionInStaticInitializer1.js

```diff
@@ -9,14 +9,13 @@ class Vector {
 }
 
 //// [thisInArrowFunctionInStaticInitializer1.js]
-var _this = this;
 function log(a) { }
 var Vector = /** @class */ (function () {
     function Vector() {
     }
     Vector.foo = function () {
         // 'this' should not be available in a static initializer.
-        log(_this);
+        log(Vector);

```

Comment:
What happens if `foo` contains a variable named `Vector`? You might want to just generate a temp variable to hold the reference.

Reply:
Ahhh.. Ok. Currently arrow function does not been a function boundary. My fault.

Reply:
That's not what I meant. Consider this scenario:

```
class Vector {
  static x = () => {
    const Vector = 1;
    return this;
  };
}
```
Since `Vector` is redeclared inside of the static block, you can't use `Vector` to access the constructor as if it were `this`. Instead I would emit the following:
```js
var Vector = (function () {
  function Vector() {
  }
  var _a = Vector;
  Vector.x = function () {
    var Vector = 1;
    return _a; // points to the outer `Vector`
  };
  return Vector;
})();
```